apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,514 @@
+        [InlineData("-Infinity")]
+        [InlineData("-inf")]
+        [InlineData("ABCDEFGH")]
+        [InlineData("1.@#$%^&*")]
+        [InlineData("1.0!")]
+        [InlineData("1.00D")]
+        [InlineData("1.00M")]
+        [InlineData("1.00F")]
+        [InlineData(".0")]
+        [InlineData(".123456")]
+        [InlineData("-.123456")]
+        [InlineData(" 1.00  ")]
+        [InlineData(" 0")]
+        [InlineData("0 ")]
+        [InlineData(" 0 ")]
+        [InlineData("+5")]
+        [InlineData(".5")]
+        [InlineData("-.5")]
+        [InlineData("+.5")]
+        [InlineData("1e6")]
+        [InlineData("1e-6")]
+        public void ParseInValidTest(string input)
+        {
+            Assert.Throws<FormatException>(() => PgNumeric.Parse(input));
+        }
+
+        [Theory]
+        [InlineData("1.00")]
+        [InlineData("1.0")]
+        [InlineData("NaN")]
+        [InlineData("1.0000")]
+        [InlineData("-1.00")]
+        [InlineData("-0.00")]
+        [InlineData("0.0000000000000009")]
+        [InlineData("-0.0000000000000009")]
+        [InlineData("1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        [InlineData("-1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        public void TryParseValidTest(string input)
+        {
+            var success = PgNumeric.TryParse(input, out PgNumeric number);
+            Assert.IsType<PgNumeric>(number);
+            Assert.NotNull(number.ToString());
+            Assert.True(success);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("       ")]
+        [InlineData("abcdefg")]
+        [InlineData("Infinity")]
+        [InlineData("inf")]
+        [InlineData("-Infinity")]
+        [InlineData("-inf")]
+        [InlineData("ABCDEFGH")]
+        [InlineData("1.@#$%^&*")]
+        [InlineData("1.0!")]
+        [InlineData("1.00D")]
+        [InlineData("1.00M")]
+        [InlineData("1.00F")]
+        [InlineData(".0")]
+        [InlineData(".123456")]
+        [InlineData("-.123456")]
+        [InlineData(" 1.00  ")]
+        [InlineData(" 0")]
+        [InlineData("0 ")]
+        [InlineData(" 0 ")]
+        [InlineData("+5")]
+        [InlineData(".5")]
+        [InlineData("-.5")]
+        [InlineData("+.5")]
+        [InlineData("1e6")]
+        [InlineData("1e-6")]
+        public void TryParseInValidTest(string input)
+        {
+            bool success = PgNumeric.TryParse(input, out PgNumeric number);
+            Assert.Equal(PgNumeric.Zero, number);
+            Assert.False(success);
+        }
+
+        [Theory]
+        [InlineData("1234,567", "es-ES")]
+        [InlineData("1.234,567", "es-ES")]
+        [InlineData("1234,567", "fr-FR")]
+        public void TryParse_UnsupportedCultures(string text, string culture)
```

Comment:
Hmmm, I don't like this tests, as they may run in parallel and this will affect the culture of all tests.
I think you can add these cases to the invalid parsing test without specifying the culture?